examples/cfq/models.py

```diff
@@ -263,12 +263,19 @@ def decode_step_fn(carry, x):
       # initialize parameters before scan
       decode_step_fn(init_carry, inputs[:, 0])
 
-    _, (logits, predictions) = jax_utils.scan_in_dim(
+    _, (logits, predictions, scores) = jax_utils.scan_in_dim(
         decode_step_fn,
         init=init_carry,  # rng, lstm_state, last_pred
         xs=inputs,
         axis=1)
-    return logits, predictions
+    if not self.is_initializing() and not train:
```

Comment:
Why did you add this `if` statement? Why not just always return the attention weights?

Reply:
I am only investigating them during evaluation, so I am avoiding unnecessary operations.

Reply:
Ok, could you add a comment explaining that this is the reason?